pyproject.toml

```diff
@@ -100,23 +94,39 @@ markers = ["unstable"]
 line-length = 99
 target-version = ["py38"]
 
-[tool.isort]
-profile = "black"
-
 # Linting tools configuration
-[tool.flake8]
-max-line-length = 99
-max-doc-length = 99
-max-complexity = 10
-exclude = [".git", "__pycache__", ".tox", "build", "dist", "*.egg_info", "venv"]
-select = ["E", "W", "F", "C", "N", "R", "D", "H"]
-# Ignore W503, E501 because using black creates errors with this
+[tool.ruff]
+target-version = "py38"
+src = ["src", "."]
+line-length = 99
+select = ["E", "W", "F", "C", "N", "D", "I001", "CPY001"]
+extend-ignore = [
+    "D203",
+    "D204",
+    "D213",
+    "D215",
+    "D400",
+    "D404",
+    "D406",
+    "D407",
+    "D408",
+    "D409",
+    "D413",
+]
+# Ignore E501 because using black creates errors with this
```

Comment:
Mostly copypaste from [ the charmcraft template](https://github.com/canonical/charmcraft/blob/main/charmcraft/templates/init-simple/pyproject.toml.j2#L17)